src/pages/Login.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, ChangeEvent } from "react";
+import styled from "@emotion/styled/macro";
+import { useAuthDispatchContext, useAuthStateContext } from "../context/auth";
+import { Redirect, RouteComponentProps } from "react-router-dom";
+import { Location } from "history";
+
+const getReferer = (location: Location) => {
```

Comment:
`referrer` should should be an optional type in `LocationState`

Reply:
`LocationState` is not a type here, but if you are referring to `Location` then that is a type defined by react router

Reply:
LocationState is a type passed to Location

Reply:
in that case it is still a type declared by react router. i also don't see why it matters...